src/main/kotlin/com/chattriggers/ctjs/api/world/TabList.kt

```diff
@@ -102,17 +123,277 @@ object TabList {
+
+        return scores.map {
+            val team = scoreboard.getTeam(it.owner)
+            TextComponent(MCTeam.decorateName(team, TextComponent(it.owner))).formattedText
+        }
+    }
+
+    /**
+     * Get all names on the tab list
+     *
+     * @return the list of names
+     */
+    @JvmStatic
+    fun getNames(): List<Name> {
+        if (needsUpdate) {
+            updateNames()
+            needsUpdate = false
+        }
+
+        return tabListNames
+    }
+
+    /**
+     * Gets all names in tabs without formatting
+     *
+     * @return the unformatted names
+     */
+    @JvmStatic
+    fun getUnformattedNames(): List<String> {
+        if (needsUpdate) {
+            updateNames()
+            needsUpdate = false
+        }
+
+        return tabListNames.map { it.toMC().profile.name }
+    }
+
+    /**
+     * Adds a new name to the tab list
+     *
+     * @param name the formatted name to add
+     * @param useExistingSkin whether to use the skin of the associated Minecraft account using [name]
+     */
+    @JvmStatic
+    @JvmOverloads
+    fun addName(name: TextComponent, useExistingSkin: Boolean = true) {
+        val connection = Client.getConnection() ?: return
+        val listedPlayerListEntries = connection.listedPlayerListEntries
+        val playerListEntries = connection.asMixin<ClientPlayNetworkHandlerAccessor>().playerListEntries
+
+        val username = name.unformattedText
+
+        val uuid = UUID.randomUUID()
+        val fakeEntry = PlayerListEntry(GameProfile(uuid, name.unformattedText), false)
+        fakeEntry.displayName = name
+
+        listedPlayerListEntries += fakeEntry
+        playerListEntries[uuid] = fakeEntry
+
+        if (!useExistingSkin) {
+            updateNames()
+            return
+        }
+
+        val mc = Client.getMinecraft()
+        val apiServices =
+            ApiServices.create(mc.asMixin<MinecraftClientAccessor>().authenticationService, mc.runDirectory)
+        apiServices.userCache.setExecutor(mc)
+
+        apiServices.userCache.findByNameAsync(username).thenAcceptAsync {
+            if (it.isPresent) {
+                val result = apiServices.sessionService.fetchProfile(it.get().id, true) ?: return@thenAcceptAsync
+
+                val entry = PlayerListEntry(result.profile, true)
+                entry.displayName = name
+
+                listedPlayerListEntries += entry
+                playerListEntries[result.profile.id] = entry
+
+                listedPlayerListEntries -= fakeEntry
+                playerListEntries.remove(uuid)
+
+                updateNames()
+            }
+        }
+    }
+
+    @JvmStatic
+    @JvmOverloads
+    fun addName(name: String, useExistingSkin: Boolean = true) = addName(TextComponent(name), useExistingSkin)
+
+    /**
+     * Removes all names from the tab list with a certain name
+     *
+     * @param name the name of the entry to remove
+     */
+    @JvmStatic
+    fun removeNames(name: TextComponent) {
```

Comment:
`removeNames` -> `removeName`. Also since you provide a `String` overload for `addName`, you should probably do the same for `removeName`.

Reply:
The issue is if there are multiple players/entries with the same names for some reason then this will remove both. I could make it remove only the first one if that is better?

Reply:
The tablist allows the same name to appear twice? That seems strange lol. I guess it's fine then